src/components/restaurantDetails/RestaurantDetails.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable jsx-a11y/img-redundant-alt */
+//Libraries
+import React from 'react';
+
+//Helpers
+import fetchRestaurantDetails from '../../api/fetchRestaurantDetails';
+import restaurantDetailsReader from './readers/restaurantDetailsReader';
+
+//Icons
+import star from '../../images/star.svg';
+import delivery from '../../images/delivery.svg';
+import rupee from '../../images/rupee.svg';
+
+//Constants
+import {
+  INITIAL_STATE,
+  RESTAURANT_ID,
+} from './constants/restaurantDetails.general';
+
+//Components
+import Offers from './components/offers';
+import Filters from './components/filters';
+import RestaurantInfo from './components/restaurantInfo';
+import Placeholder from './components/placeholder';
+import Loader from '../../commonComponents/loader';
+import ErrorHandler from '../../commonComponents/errorHandler';
+
+//CSS
+import './restaurantDetails.css';
+
+class RestaurantDetails extends React.Component {
+  state = INITIAL_STATE;
+
+  componentDidMount() {
+    this.getRestaurantDetails();
+  }
+
+  getRestaurantDetails() {
+    fetchRestaurantDetails(RESTAURANT_ID)
+      .then(this.setRestaurantDetails)
+      .catch(this.handleError)
+      .finally(this.setAsLoaded);
+  }
+
+  setRestaurantDetails = (restaurantDetails) => {
+    this.setState({
+      restaurantDetails: restaurantDetails,
+    });
+  };
+
+  handleError = (error) => {
+    const { message = 'Something went wrong' } = error;
```

Comment:
You can have 'Something went wrong' somewhere common and use the constant everywhere needed, like default value in Error component